src/pudl/analysis/pudl_models.py

```diff
@@ -48,7 +51,179 @@ def core_sec10k__quarterly_company_information() -> pd.DataFrame:
 
 @asset(
     io_manager_key="pudl_io_manager",
-    group_name="pudl_models",
+    group_name="sec10k",
+)
+def core_sec10k__quarterly_company_information(
+    raw_sec10k__quarterly_company_information: pd.DataFrame,
+) -> pd.DataFrame:
+    """Company information extracted from SEC10k filings.
+
+    Consolidate company information extracted from key: value blocks
+    within the headers of the SEC 10k filings such that there
+    is one record of company information per block. One company's
+    information may be reported in multiple filings on the same report date.
+    However, we only want to keep one of those records per company and report date,
+    (``central index key`` and ``report_date`` are a primary key for the table).
+    We prioritize keeping records from filings where that company's extracted
+    ``central_index_key`` matches the filer's central index key, meaning that
+    that company filed the 10k itself.
+    """
+    # Strip erroneous "]" characters
+    raw_sec10k__quarterly_company_information["company_information_fact_name"] = (
+        raw_sec10k__quarterly_company_information[
+            "company_information_fact_name"
+        ].str.lstrip("]")
+    )
+    df = raw_sec10k__quarterly_company_information.pivot(
+        values="company_information_fact_value",
+        index=[
+            "filename_sec10k",
+            "report_date",
+            "company_information_block",
+            "company_information_block_count",
+        ],
+        columns="company_information_fact_name",
+    )
+    df.columns.name = None
+    df = df.reset_index()
+    # consolidate information extracted from blocks within the header
+    # so that there is one record per block
+    df = (
+        (
+            df.groupby(
+                ["filename_sec10k", "report_date", "company_information_block_count"]
+            ).first()
+        )
+        .reset_index()
+        .drop(columns=["company_information_block", "company_information_block_count"])
+        .dropna(subset="central_index_key")
+    )
+    # we want central_index_key and report_date to be a primary key
+    # prioritize records where the filer is the same
+    # as the harvested central index key value
+    df["filer_cik"] = df["filename_sec10k"].str.split("/").str[2].str.zfill(10)
+    df["filer_cik_matches_cik"] = df["filer_cik"] == df["central_index_key"]
+    df = df.sort_values(by="filer_cik_matches_cik", ascending=False).drop_duplicates(
+        subset=["central_index_key", "report_date"], keep="first"
+    )
```

Comment:
Interestingly, if you don't do this sorting before dropping duplicates on CIK and report dates then there are thousands of duplicate filenames + report dates. But when you do this sorting first there are only 2 duplicate filename + report dates. This felt wrong to me, but I couldn't figure out why it would be wrong. I think it's actually good - it means that almost all company information can be harvested directly from a filing submitted by the company itself.